hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/avro/HoodieSparkSchemaConverters.scala

```diff
@@ -102,7 +102,14 @@ object HoodieSparkSchemaConverters {
           val fields = st.map { f =>
             val fieldSchema = toHoodieType(f.dataType, f.nullable, f.name, childNameSpace)
             val doc = f.getComment.orNull
-            HoodieSchemaField.of(f.name, fieldSchema, doc)
+            // Match existing Avro SchemaConverters behavior: use NULL_VALUE for nullable unions
+            // to avoid serializing "default":null in JSON representation
+            val defaultVal = if (fieldSchema.isNullable) {
+              HoodieJsonProperties.NULL_VALUE
+            } else {
+              null
+            }
+            HoodieSchemaField.of(f.name, fieldSchema, doc, defaultVal)
           }
```

Comment:
On line 190 there is a bug is this code where the comment is added even if the doc string is empty which deviates from the avro path. That is causing some of the failures in CI

Reply:
thanks for catching this, will fix

Reply:
Have made a change on that line 190 let me know if it makes sense